app/src/main/java/com/liato/bankdroid/DataRetrieverTask.java

```diff
@@ -68,37 +72,74 @@ public DataRetrieverTask(final MainActivity parent, final long bankId) {
 
     @Override
     protected void onPreExecute() {
-        this.dialog.setMessage(res.getText(R.string.updating_account_balance)
+        getDialog().setMessage(res.getText(R.string.updating_account_balance)
                 + "\n ");
-        this.dialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
-        this.dialog.setCancelable(false);
-        this.dialog.show();
+        getDialog().setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
+        getDialog().setCancelable(false);
+        getDialog().show();
+    }
+
+    @NonNull
+    protected ProgressDialog getDialog() {
+        return this.dialog;
+    }
+
+    protected Bank getBankFromDb(long bankId, Context parent, boolean loadAccounts) {
+        return BankFactory.bankFromDb(bankId, parent, true);
```

Comment:
The parameter `loadAccounts` is never used in the method. Please either pass it on to the `BankFactory#bankFromDb` or remove it from the parameter list.